llvm/include/llvm/IR/ProfDataUtils.h

```diff
@@ -235,8 +239,8 @@ getDisjunctionWeights(const SmallVector<uint32_t, 2> &B1,
   // the product of sums, the subtracted one cancels out).
   assert(B1.size() == 2);
   assert(B2.size() == 2);
-  auto FalseWeight = B1[1] * B2[1];
-  auto TrueWeight = B1[0] * B2[0] + B1[0] * B2[1] + B1[1] * B2[0];
+  uint64_t FalseWeight = B1[1] * B2[1];
```

Comment:
What's the motivation for changing this? Previously it would default to the integer type, but now it's fixed.

Shouldn't be a big issue for `uint64_t` specifically, but maybe a good idea to add a static assert that the bit width of `T1`/`T2` is less than or equal to 64?